doc/develop/test/twister.rst

```diff
@@ -375,6 +375,25 @@ harness: <string>
     Usually pertains to external dependency domains but can be anything such as
     console, sensor, net, keyboard, Bluetooth or pytest.
 
+platform_key: <list of platform attributes>
+    Often a test needs to only be built and run once to qualify as passing.
+    Imagine a library of code that depends on the platform architecture where
+    passing the test on a single platform for each arch is enough to qualify the
+    tests and code as passing. The platform_key attribute enables doing just
+    that.
+
+    For example to key on (arch, simulation) to ensure a test is run once
+    per arch and simulation (as would be most common)::
+
+      platform_key:
+        - arch
+        - simulation
```

Comment:
can we have somewhere to defined valid list for platform_key? this could be abused.

Reply:
This is an interesting point, I don't limit it currently. Its any (currently) top level attribute of a board yaml. Do you have a particular set of board/platform attributes you'd like to limit this to?

Reply:
I do not have a defined list yet, but maybe we can reuse what we are discussing here https://github.com/zephyrproject-rtos/zephyr/issues/51718